agents/xy_stage/xy_ocs_agent.py

```diff
@@ -0,0 +1,249 @@
+                return False, "could not acquire lock"
+            self.xy_stage.move_y_cm( params['distance'], params['velocity'])
+        
+        time.sleep(1)
+        while True:
+            ## data acquisition updates the moving field if it is running
+            if not self.take_data:
+                with self.lock.acquire_timeout(timeout=3, job='move_y_cm') as acquired:
+                    if not acquired:
+                        self.log.warn(f"Could not check for move because lock held by {self.lock.job}")
+                        return False, "could not acquire lock"
+                    self.is_moving = self.xy_stage.moving
+            if not self.is_moving:
+                break
+        return True, "Y Move Complete"
+
+ 
+    def set_position(self, session, params):
+        """
+        params: dict: {'position': (float, float)}
+        """
+        with self.lock.acquire_timeout(timeout=3, job='set_position') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not set position because lock held by {self.lock.job}")
+                return False, "Could not acquire lock"
+                        
+            self.xy_stage.position = params['position']
+        return True, "Position Updated"
+
+    def start_acq(self, session, params=None):
+        """
+        params: dict: {`sampling_frequency': float, sampling rate in Hz}
+        """
+        if params is None:
+            params = {}
+
+        
+        f_sample = params.get('sampling_frequency', self.sampling_frequency)
+        pm = Pacemaker(f_sample, quantize=True)
+
+        if not self.initialized:
+            self.init_xy_stage_task(session)
+        
+        with self.lock.acquire_timeout(timeout=0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            self.log.info(f"Starting Data Acquisition for XY Stages at {f_sample} Hz")
+            session.set_status('running')
+            self.take_data = True
+            last_release = time.time()
+
+            while self.take_data:
+                if time.time()-last_release > 1.:
+                    if not self.lock.release_and_acquire(timeout=10):
+                        self.log.warn(f"Could not re-acquire lock now held by {self.lock.job}.")
+                        return False, "could not re-acquire lock"
+                    last_release = time.time()
+                pm.sleep()
+
+                data = {'timestamp':time.time(), 'block_name':'positions','data':{}}
+                pos = self.xy_stage.position
+                self.is_moving = self.xy_stage.moving
+
+                data['data']['x'] = pos[0]
+                data['data']['y'] = pos[1] 
+                self.agent.publish_to_feed('positions',data)
+                self.agent.feeds['positions'].flush_buffer()
```

Comment:
Instead of flushing the ocs feed every sample, you can set `buffer_time=0` when registering the feed instead of `buffer_time=1`.  That will make it so data is sent as soon as the publish statement is called.